optuna/integration/chainermn.py

```diff
@@ -168,157 +166,190 @@ def __setattr__(self, attr_name, value):
         setattr(self.delegate, attr_name, value)
 
 
-class _ChainerMNPruner(BasePruner):
-    def __init__(self, pruner, comm):
-        # type: (BasePruner, CommunicatorBase) -> None
-
-        self.delegate = pruner
-        self.comm = comm
-
-    def prune(self, storage, study_id, trial_id, step):
-        # type: (BaseStorage, int, int, int) -> bool
-
-        if self.comm.rank == 0:
-            try:
-                result = self.delegate.prune(storage, study_id, trial_id, step)
-                self.comm.mpi_comm.bcast(result)
-                return result
-            except Exception as e:
-                self.comm.mpi_comm.bcast(e)
-                raise
-        else:
-            result = self.comm.mpi_comm.bcast(None)
-            if isinstance(result, Exception):
-                raise result
-            return result
-
-
-class _ChainerMNTrial(Trial):
+class ChainerMNTrial(BaseTrial):
```

Comment:
In this PR, `ChainerMNTrial` will be a public class, and users may want to consult its documentation when they write objective functions.
Could you add documentation for `ChainerMNTrial`?

The following is just an example:
```
    """A wrapper of :class:`~optuna.trial.Trial` to incorporate Optuna with ChainerMN.

    .. seealso::
        :class:`~optuna.integration.chainermn.ChainerMNTrial` provides the same interface as
        :class:`~optuna.trial.Trial`. Please refer to :class:`optuna.trial.Trial` for further
        details.

    Args:
        trial:
            A :class:`~optuna.trial.Trial` object.
        comm:
            A `ChainerMN communicator <https://docs.chainer.org/en/stable/chainermn/reference/
            index.html#communicators>`_.
    """

```

Reply:
Added: https://github.com/pfnet/optuna/pull/427/commits/e67caf1fe67ef3982fbfcd47243cbeb5cd28e6b4

Thank you for your suggestion.